### ISSUE TITLE
Fuzzy fonts.

### ISSUE THREAD

**Issue by se5a** [Pulsar4xDevs/Pulsar4x#292]
Currently we're using the IMGui default ProggyClean.ttf because it's nice and crisp at a small (13) size. 
every other font I've tried is fuzzy, and most don't render well at small sizes.
the problem with ProggyClean however, is that it is missing some special chars I'd like to use in a few places, eg: ΩωνΔθ.
I can add those chars from another font, but when used in a string, these chars will cause the rest of the string to be fuzzy: 
![image](https://user-images.githubusercontent.com/1490538/85346051-5e255880-b548-11ea-8e34-4f820d2e847a.png)

if it was just the special char it wouldn't be so bad.

**Comment by se5a** [comment]
Could we edit the ProggyClean ttf and add the extra chars that way?

**Comment by se5a** [comment]
Aaand the problem with text following these special chars appears to have (at least mostly) gone away:
![image](https://user-images.githubusercontent.com/1490538/89110239-e6c6db00-d49c-11ea-828d-1ce2d2da3014.png)

I noticed this after adding a handful more chars (ie ΩωΝνΔδθΘ);
so I assumed that the number of chars added (via `builder.AddText("ΩωΝνΔδθΘ");`) had affected this. finaly got around to sitting down and doing some tests to prove this to be the case, and I can no longer replicate the problem shown in the "Total Δv for all manuvers: 0 m/s" text line in the screenshot in the OP.
I've gone back to the original five chars (ΩωνΔθ) for the screenshot in this comment.  
I'm now stumped, only thing I can think of now is maybe it is/was a driver issue, but that seems a little unlikely. 

Edit notes:
- Omar suggests we try "Using freetype with hinting, or subpixel rendering" for small size fonts. 
- I'm also noticing the v after the Delta symbol still appears fuzzy. 
- I should probably get the repo at the June23 date, and see if I can replicate it with that, should be able to rule out or in whether its our code or something external like a video driver update.

**Comment by se5a** [comment]
Tried the repo at 533fe6c8 and it looks similar to the second screenshot (ie the text after the special char is not fuzzy). starting to think maybe it might have been a driver issue.

**Comment by se5a** [comment]
Tried rolling back video drivers. appeared to have no effect. 
I'm noticing in the screenshots above that the text on the "High Δv Intercept" button is still fuzzy though. 
godamnit, this is frustrating.  what's causing this to be a problem sometimes, but not others?

**Comment by se5a** [comment]
Ok so to consolidate the things I've learned:
- This issue is caused by... the glyph (widths?) not being whole numbers, the "AdvanceX" value for a given glyph shows this. 
- The an AdvanceX value can be viewed in the imgui demo->Tools->StyleEditor:
![image](https://user-images.githubusercontent.com/1490538/89118748-44d6db00-d4fc-11ea-8325-87a490c25530.png)
You can see in the above image the tooltip showing the AdvanceX for one of the DejaVuSans glyphs. 
- All the proggy clean glyphs show an AdvanceX of 7.0 (a nice whole number). 
- Setting PixelSnapH to true in the config when merging the glyphs is supposed to fix this issue, however as the image above shows, the dejavusans font, the PixelSnapH is set to 1 (true) however the AdvanceX value is still not a whole number.

Relevent code:
```cs
unsafe
{
    string rf = "Resources";
    
    ImFontConfig* rawPtr = ImGuiNative.ImFontConfig_ImFontConfig();
    ImFontConfigPtr config = new ImFontConfigPtr(rawPtr);
    config.PixelSnapH = true;
    config.MergeMode = true;
    
    ImFontAtlasPtr fontAtlas = ImGui.GetIO().Fonts;
    var builder = new ImFontGlyphRangesBuilderPtr(ImGuiNative.ImFontGlyphRangesBuilder_ImFontGlyphRangesBuilder());
    builder.AddText("ΩωΝνΔδθΘ"); //Omega, Nu, Delta, Theta (UPPER and lower cases)
    //builder.AddText("ΩωνΔθ");
    //builder.AddRanges(fontAtlas.GetGlyphRangesDefault());
    builder.BuildRanges(out ImVector ranges);
    fontAtlas.AddFontFromFileTTF(Path.Combine(rf, "ProggyClean.ttf"), 13);
    fontAtlas.AddFontFromFileTTF(Path.Combine(rf, "DejaVuSans.ttf"), 13, config, ranges.Data);
    
}
```

**Comment by se5a** [comment]
Fixed!
![image](https://user-images.githubusercontent.com/1490538/89120690-69d34a00-d50c-11ea-8546-ea6b3f00ace3.png)

That looks *way* better.
Ended up there is a bug in imgui where it's using the PixelSnapH from the first font loaded, rather from the config.
it also was a little bit not obvious what was wrong since if proggy clean is loaded via  AddFontDefault(), it sets the PixelSnapH to true. (if I'd been loading proggyclean that way it the config as per my previous comment would have worked)
ie:
```
config.PixelSnapH = true;
config.MergeMode = true;
fontAtlas.AddFontDefault();
fontAtlas.AddFontFromFileTTF(Path.Combine(rf, "DejaVuSans.ttf"), 13, config, ranges.Data);
```
alternativly, we can pass the config when loading the first font:
```
config.PixelSnapH = true;
fontAtlas.AddFontFromFileTTF(Path.Combine(rf, "ProggyClean.ttf"), 13);
config.MergeMode = true;
fontAtlas.AddFontFromFileTTF(Path.Combine(rf, "DejaVuSans.ttf"), 13, config, ranges.Data);
```
which is how I've fixed it in the last commit. note also I needed to set MergeMode *after* I'd loaded the first font or I got a silent crash. 

Thanks to Omar for his help with this one.
